### ISSUE TITLE
No Match found

### ISSUE THREAD

**Issue by wuyangang** [coding-motion/ionic-keycloak#2]
Prompt No Match found after app login，why?

**Comment by wieczorekmarcin** [comment]
I created AuthGuard (implementing CanActivate). After logging out, I am redirected back to the keycloak login form, then I am redirected to the application where I receive the same error "No Match found" and all requests are unauthorized. After turning the application off and on, everything works fine until I log out.

**Comment by Rubliev** [comment]
I had the same problem. It seems related to the deeplink (ad documented here https://github.com/ionic-team/ionic-plugin-deeplinks/issues/77 ), apparently the observable it's unsubscribed after the first call.

In my project I modified the init method in the deep-link.service into something like this. It's working, even if it's just a workaround.


```
init() {
    this.platform.ready().then(() => {
      if (window['cordova']) {
        const dl = window['IonicDeeplink'];
        dl.route(
            {
            },
            (match) => {
              // match.$route - the route we matched, which is the matched entry from the arguments to route()
              // match.$args - the args passed in the link
              // match.$link - the full link data
              this.extractData(match.$link);
            },
            (nomatch) => {
              // nomatch.$link - the full link data
              this.extractData(nomatch.$link);
            }
        );
      }
    });
  }
```

Not the prettiest of the solutions. I hope it helps

**Comment by SheppardX** [comment]
I had the same error, but for me the keycloak was the problem ... i got an XML Request error and cors from it ... are u sure u configurated keycloak correctly ?

**Comment by daankennes** [comment]
> I created AuthGuard (implementing CanActivate). After logging out, I am redirected back to the keycloak login form, then I am redirected to the application where I receive the same error "No Match found" and all requests are unauthorized. After turning the application off and on, everything works fine until I log out.

Same problem. If I click register first and login after that, it works again.

**Comment by bodronRamona** [comment]
Hello, Did you find a solution?

**Comment by daankennes** [comment]
Nope, just removed it from my project. But it's not very difficult to program this yourself I believe.
